.github/workflows/postman-collection-runner.yml

```diff
@@ -97,23 +97,37 @@ jobs:
         env:
           DATABASE_URL: postgres://db_user:db_pass@localhost:5432/hyperswitch_db
         run: diesel migration run
-
+      
+      - name: Build project
+        if: ${{ ((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) || (github.event_name == 'merge_group')}}
+        run: cargo build
+  
       - name: Setup Local Server
         if: ${{ ((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) || (github.event_name == 'merge_group')}}
         run: |
+          # Check if the build succeeded
+          if [ $? -ne 0 ]; then
+            echo "Cargo build failed. Exiting."
+            exit 1
+          fi
```

Comment:
This is not necessary, since the workflow would stop if any previous step fails.